src/module/generate-routes.ts

```diff
@@ -53,3 +53,28 @@ export const generateRoutes = async (
 
   return metadata;
 };
+
+const exactly = (input: SwaggerConfigRelatedToRoutes): SwaggerConfigRelatedToRoutes => {
+  // Validate the config values first
+  if (
+    input.noImplicitAdditionalProperties === undefined ||
+    input.noImplicitAdditionalProperties === 'throw-on-extras' ||
+    input.noImplicitAdditionalProperties === 'silently-remove-extras'
+  ) {
+    // then it's good to go
+  } else {
+    throw new Error(`noImplicitAdditionalProperties is set to an invalid value. See https://github.com/lukeautry/tsoa/blob/master/src/config.ts for available options.`);
```

Comment:
@endor I suppose it is a "breaking change" that I'm no longer accepting `true` or `false` as possible values, but:
(a) the code got confusing and gross if I were to allow booleans
(b) I couldn't determine which of the two "modes" (`'throw-on-extras'` or `'silently-remove-extras'`) would `true` default to?
(c) I felt that since the 2.4.4 release just went out this morning that it would be okay to make a breaking change when it's probably very unlikely that anyone has installed the new version and utilized this un-advertised feature. That being said, if they were to upgrade to this code they would at least get this (hopefully) very helpful error message.